bbc/simorgh
# WSTEAM1-1137: Move focus to basic 'error summary' box

Resolves JIRA [WSTEAM1-1137](https://jira.dev.bbc.co.uk/browse/WSTEAM1-1137) - Move focus to first error message

Overall changes
======
After a user submits a form with validation errors
- Renders a basic* error summary box at top of form
- Takes user's focus to box
- Updates page title

After the user resolves errors:
- Removes basic* error summary box from top of form
- Updates page title

(*Basic in that it does not contain a list of errors. Instead it contains a strong element indicating there are validation errors).

Code changes
======
- Repurposes InavlidMessageBox
- Imports InvalidMessageBox into Form Screen
- Adds useEffect to FormScreen to handle page Title change
- Updates FormContext handler - hasAttemptedSubmit becomes attemptedSubmitCount. Adds hasValidationErrors.
- Updates styling naming
- Adds unit tests

Note
- Changes InvalidMessageBox paragraph element to strong element to match screen reader UX. (Assuming this was missed)

Testing
======
1. Pull down this PR
2. Visit http://localhost.bbc.com:7081/mundo/send/u50853489?renderer_env=live or http://localhost.bbc.com:7081/somali/send/u130092370?renderer_env=test
3. Press submit on an invalid form, check focus moves to the Error box
4. Resolve errors, check error summary box disappears
5. Reintroduce errors, check box reappears
6. Press submit multiple times with and without making changes and ensure focus returns to the error summary box each time,
7. Check it matches the Screen reader UX
8. Check it matches the A11y AAC's

Helpful Links
======
AAC specified on subtask: https://jira.dev.bbc.co.uk/browse/WSTEAM1-1156
Figma designs:
Design for uploading and completed - https://www.figma.com/design/hbI3T5RehCO1EUsb1p9aI3/Uploader-update---MVP---Handoff-file?node-id=2003-17730&t=4txyX996EDTlnhgf-4
Interaction states for buttons and links - https://www.figma.com/design/hbI3T5RehCO1EUsb1p9aI3/Uploader-update---MVP---Handoff-file?node-id=2003-3588&t=tgfjJJxyWLhymkko-0 
AAC: https://paper.dropbox.com/doc/Uploader-Accessibility-acceptance-criteria--COvw5wlY_TTWnVuWywBxP8fmAg-BN1tII16uRaecL0YQQ0pR#:uid=210887839769911648141335&h2=All-stages 
A11y feedback doc: https://paper.dropbox.com/doc/Uploader-UX-A11y--COQaWVQTblzBBbtkIYJDCH8FAg-tFXBb8Ic07uVAebveWjbL
Screen reader UX: https://paper.dropbox.com/doc/WS-Uploader-MVP-Screenreader-UX--CSmteeUNHj_ghnhgdB5400ClAg-tgOv3vS52cfR5tpaqmJiI

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
